lib/board.rb

```diff
@@ -0,0 +1,60 @@
+class Board
+  attr_accessor :cells
+
+  def initialize
+    @cells = []
+    reset!
+  end
+
+  def reset!
+    @cells.clear
+    @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end  
+
+  def display
+    puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+    puts "-----------"
+    puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+    puts "-----------"
+    puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+  end
+
+  def position(number)
+    cells[number.to_i-1]
+  end
+
+  def full?
+    spot_filled = []
+    cells.each do |cell| 
+      if cell.include?("X") || cell.include?("O")
+        spot_filled << true
+      else 
+        spot_filled << false
+      end
+    end
+    ( spot_filled.include?(false) ? false : true ) 
+  end
+
+  def turn_count
+    turn_total = 0
+    cells.each do |cell| 
+      turn_total += 1 if cell.include?("X") || cell.include?("O")
+    end
+    turn_total
+  end
```

Comment:
For turn_count, the [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) iterator on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count {|cell| cell == "X" || cell == "O"}
end
```